File: Arrays/Easy/1773/1773.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        output: int = 0
        for item in items:
            if item[0] == ruleValue and ruleKey == "type": 
                output +=1
            elif item[1] == ruleValue and ruleKey == "color": 
                output +=1
            elif item[2] == ruleValue and ruleKey == "name": 
                output +=1
        return output

File: Arrays/Easy/1773/1773.ts
function countMatches(
  items: string[][],
  ruleKey: string,
  ruleValue: string
): number {
  let output: number = 0;
  const map: { [key: string]: number } = {
    type: 0,
    color: 1,
    name: 2,
  };

  const index = map[ruleKey];

  for (const item of items) {
    if (item[index] === ruleValue) {
      output++;
    }
  }

  return output;
}
